src/transformers/tokenization_xlm_roberta.py

```diff
@@ -261,21 +269,40 @@ def create_token_type_ids_from_sequences(
 
     @property
     def vocab_size(self):
-        return len(self.sp_model) + len(self.fairseq_tokens_to_ids)
+        return len(self.sp_model) + self.fairseq_offset + 1  # Add the <mask> token
```

Comment:
I really dislike this. This assumes that there is no `<mask>` token in the vocabulary, and adds it. Furthermore, the mask token is NOT in the vocabulary, while all the other tokens that are in `self.fairseq_tokens_to_ids` ARE in the vocabulary.

This is not robust, and would probably break were we to load a SPM with an actual mask token.

Reply:
Totally agree. Once we have sentencepiece support in `tokenizers` (cc @n1t0 :-) ) we might be able to update the sentence piece vocabularies themselves to clean up these mixing of python code and sentencepiece code.

Reply:
Updating the SPM files themselves could definitely work (and I believe we could do it right now, but it requires diving deep in SentencePiece), but it would also mean that we wouldn't be able to convert new XLM-R models (if there are to be any). I guess it's a tradeoff:

- **Change the SPM file so that the implementation is clean**: clean and intuitive code with no messy workarounds, but can't handle new XLM-R models without manually updating their SPM files.
- **Don't change the SPM file**: implementation stays as it is, extremely bound to the current files output by the XLM-R team, therefore probably nothing to do if new XLM-R models ever come out